### PR TITLE
Fix a couple of client-side redirects

### PR DESCRIPTION
**Pull request by cnunciato**
Following up on https://github.com/pulumi/pulumi-hugo/pull/1013#pullrequestreview-867284943, I noticed a couple of errors in the client-side redirects that were added in https://github.com/pulumi/pulumi-hugo/pull/938. This change should fix them up.

### CODE REVIEW

File: themes/default/content/docs/intro/concepts/resources/_index.md [hunk 1]
```diff
@@ -28,13 +28,13 @@ menu:
         "#version": "/docs/intro/concepts/resources/options/version",
         "#components": "/docs/intro/concepts/resources/components",
         "#authoring-a-new-component-resource": "/docs/intro/concepts/resources/components/#authoring-a-new-component-resource",
-        "#creating-a-child-resource": "/docs/intro/concepts/resources/components/#creating-a-child-resource",
+        "#creating-child-resources": "/docs/intro/concepts/resources/components/#creating-child-resources",
```
Comment (cnunciato) [inline themes/default/content/docs/intro/concepts/resources/_index.md:31]: The [previous version of this page](https://raw.githubusercontent.com/pulumi/pulumi-hugo/12c547508213699a93cde3aae460f610e21d6a24/themes/default/content/docs/intro/concepts/resources.md) used the heading `Creating Child Resources`, so the resulting anchor tag would've been `creating-child-resources`.

File: themes/default/content/docs/intro/concepts/resources/_index.md [hunk 2]
```diff
@@ -28,13 +28,13 @@ menu:
...
         "#registering-component-outputs": "/docs/intro/concepts/resources/components/#registering-component-outputs",
         "#inheriting-resource-providers": "/docs/intro/concepts/resources/components/#inheriting-resource-providers",
         "#providers": "/docs/intro/concepts/resources/providers",
         "#default-provider-configuration": "/docs/intro/concepts/resources/providers/#default-provider-configuration",
         "#explicit-provider-configuration": "/docs/intro/concepts/resources/providers/#explicit-provider-configuration",
-        "#dynamicproviders": "/docs/intro/concepts/resources/dynamicproviders",
+        "#dynamicproviders": "/docs/intro/concepts/resources/dynamic-providers",
```
Comment (cnunciato) [inline themes/default/content/docs/intro/concepts/resources/_index.md:37]: Note that for this heading, [we applied a custom anchor](https://raw.githubusercontent.com/pulumi/pulumi-hugo/12c547508213699a93cde3aae460f610e21d6a24/themes/default/content/docs/intro/concepts/resources.md) (`dynamicproviders`) -- I'm not sure why. 🤷

**Review by lukehoban — APPROVED** [review]
Thanks for fixing these up - sorry for creating these issues in the first place!

**Review by susanev — APPROVED** [review]
thank you thank you!

**Comment by cnunciato** [comment]
Not at all -- I should've taken the time to check these individually as well when I reviewed the PR. 🤦

**Pull request merged**